### PR TITLE
stats: fix combined index low-bound check

### LINKED ISSUE #7813: combined index low bound check question
## Bug Report

1. What did you do?
If possible, provide a recipe for reproducing the error.

```
create table t(id int auto_increment, kid int, pid int, primary key(id), key(kid, pid));
insert into t (kid, pid) values (1,2), (1,3), (1,4),(1, 11), (1, 12), (1, 13), (1, 14), (2, 2), (2, 3), (2, 4);
analyze table t;
```
then show bucket will be this:

```
mysql> SHOW STATS_BUCKETS where column_name = 'kid';                                                                                                                                               +---------+------------+-------------+----------+-----------+-------+---------+-------------+-------------+
| Db_name | Table_name | Column_name | Is_index | Bucket_id | Count | Repeats | Lower_Bound | Upper_Bound |
+---------+------------+-------------+----------+-----------+-------+---------+-------------+-------------+
| test    | t          | kid         |        1 |         0 |     1 |       1 | (1, 2)      | (1, 2)      |
| test    | t          | kid         |        1 |         1 |     2 |       1 | (1, 3)      | (1, 3)      |
| test    | t          | kid         |        1 |         2 |     3 |       1 | (1, 4)      | (1, 4)      |
| test    | t          | kid         |        1 |         3 |     4 |       1 | (1, 11)     | (1, 11)     |
| test    | t          | kid         |        1 |         4 |     5 |       1 | (1, 12)     | (1, 12)     |
| test    | t          | kid         |        1 |         5 |     6 |       1 | (1, 13)     | (1, 13)     |
| test    | t          | kid         |        1 |         6 |     7 |       1 | (1, 14)     | (1, 14)     |
| test    | t          | kid         |        1 |         7 |     8 |       1 | (2, 2)      | (2, 2)      |
| test    | t          | kid         |        1 |         8 |     9 |       1 | (2, 3)      | (2, 3)      |
| test    | t          | kid         |        1 |         9 |    10 |       1 | (2, 4)      | (2, 4)      |
+---------+------------+-------------+----------+-----------+-------+---------+-------------+-------------+
10 rows in set (0.01 sec)

```

now 

```
explain select * from t where kid = 1;
```

*this question only appear `=1`*

2. What did you expect to see?

```
explain select * from t where kid = 1;
+-------------------+-------+------+--------------------------------------------------------+
| id                | count | task | operator info                                          |
+-------------------+-------+------+--------------------------------------------------------+
| IndexReader_9     | 7.00  | root | index:IndexScan_8                                      |
| └─IndexScan_8     | 7.00  | cop  | table:t, index:kid, pid, range:[1,1], keep order:false |
+-------------------+-------+------+--------------------------------------------------------+

```

3. What did you see instead?

```
explain select * from t where kid = 1;
+-------------------+-------+------+--------------------------------------------------------+
| id                | count | task | operator info                                          |
+-------------------+-------+------+--------------------------------------------------------+
| IndexReader_9     | 0.00  | root | index:IndexScan_8                                      |
| └─IndexScan_8     | 0.00  | cop  | table:t, index:kid, pid, range:[1,1], keep order:false |
+-------------------+-------+------+--------------------------------------------------------+
2 rows in set (0.00 sec)
```

4. What version of TiDB are you using (`tidb-server -V` or run `select tidb_version();` on TiDB)?

```
2.1.0.rc2
```

### PR DESCRIPTION
**Pull request by lysu**
### What problem does this PR solve? 

fixes #7813

`=1` will failure in low-bound check and break, so can not do following cm-sketch logic.

### What is changed and how it works?

for range `(kid1, pid2) ~ (kid2,pid2)`, judge `kid1` as hit in range.

### Check List 

Tests 

 - unit test

Code changes

 - logic change

Side effects

 - no

Related changes

 - Need to cherry-pick to the release branch

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tidb/7814)

### CODE REVIEW

File: util/chunk/compare.go [hunk 1]
```diff
@@ -163,6 +164,11 @@ func cmpJSON(l Row, lCol int, r Row, rCol int) int {
 
 // Compare compares the value with ad.
 func Compare(row Row, colIdx int, ad *types.Datum) int {
+	return CompareOpt(row, colIdx, ad, false)
+}
+
+// CompareOpt compares the value with ad with prefixAsEq optional
+func CompareOpt(row Row, colIdx int, ad *types.Datum, prefixAsEq bool) int {
```
Comment (shenli) [inline util/chunk/compare.go]: Please add comments for `prefixAsEq`.

Comment (lysu) [reply util/chunk/compare.go]: emmm..this param is stranger, I tried to refactor it

Comment (Apollo1993) [reply util/chunk/compare.go]: prefixAsEq

**Pull request merged**